docs/proposed/http-introducer.rst

```diff
@@ -0,0 +1,153 @@
+.. -*- coding: utf-8 -*-
+
+HTTP Introducer
+===============
+
+This document explains use and operation of the "HTTP Introducer" system.
+This system replaces the earlier "Introducer node" system.
+
+The "HTTP Introducer" system is a means for client nodes to initially learn about available storage nodes.
+It also allows client nodes to receive updates to this information.
+The goal of this document is to explain how to *use* this system,
+either as a user of a Tahoe-LAFS client node or as the operator of a Tahoe-LAFS grid or storage node.
+
+
+Use As A Client
+---------------
+
+You want your Tahoe-LAFS client to be able to find storage servers to access.
+This means your client needs both current and future connection information for those storage servers.
+By configuring your client with one small static connection hint,
+the "HTTP Introducer" system enables the client to find more information and future updates to that information.
+
+Before attempting this configuration,
+you should have an HTTP-style *introducer fURL* from the operator of an HTTP Introducer-enabled Tahoe-LAFS storage grid.
+
+The HTTP Introducer is configured in two parts.
+First,
+the *introducer fURL* is written to a new file.
+This configuration should be treated as a secret.
+The node's ``private`` directory is a good location to consider for this file.
+For example,
+for a grid you refer to as the "foo grid" you might use ``private/foogrid-introducer``.
+
+Next,
+in ``tahoe.cfg`` the ``grid-introducer-path`` item in the ``[client]`` section is set to refer to this file.
+For example::
+
+  [client]
+  http-introducer-path = private/foogrid-introduction
```

Comment:
:+1: this is good, avoids (more) seekrits in the config-file.

Reply:
Perhaps even the user shouldn't get a choice and they put e.g. `foogrid-introduction` in the config, and it's in `private/foogrid-introduction` (it could also support absolute paths if we want). I guess all I'm saying here is that maybe relative paths should be relative to `<node-dir>/private/` instead of `<node-dir>/`

Reply:
Turns out we already have `private/introducers.yaml`!  So I put up #897 to just finally get rid of introducer config in `tahoe.cfg`.